test/suite/test_checkpoint31.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import threading, time
+import wttest
+import wiredtiger
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+from wiredtiger import stat
+
+# test_checkpoint31.py
+#
+# Test that skipping in-memory reconciled deleted pages as part of the tree walk.
+
+class test_checkpoint(wttest.WiredTigerTestCase):
+    conn_config = 'cache_size=500MB,statistics=(all)'
+
+    format_values = [
+    # FLCS doesn't support skipping pages based on aggregated time window.
+    #    ('column-fix', dict(key_format='r', value_format='8t',
+    #        extraconfig=',allocation_size=512,leaf_page_max=512')),
+        ('column', dict(key_format='r', value_format='S',
+            extraconfig=',allocation_size=512,leaf_page_max=512')),
+        ('string_row', dict(key_format='S', value_format='S',
+             extraconfig=',allocation_size=512,leaf_page_max=512')),
+    ]
+    
+    scenarios = make_scenarios(format_values)
+
+    def check(self, ds, nrows, value):
+        cursor = self.session.open_cursor(ds.uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, value)
+            count += 1
+        self.assertEqual(count, nrows)
+        cursor.close()
+
+    def test_checkpoint(self):
```

Comment:
Perhaps we could change the name of this test?

Reply:
Changed into test_evict01, as I feel evict is mostly related to this test.